src/lib/mdns/Advertiser_ImplMinimalMdns.cpp

```diff
@@ -608,9 +608,9 @@ FullQName AdvertiserMinMdns::GetCommisioningTextEntries(const CommissionAdvertis
         }
 
         char txtPairingInstr[chip::Mdns::kKeyPairingInstructionMaxLength + 4];
-        if (params.GetPairingInstr().HasValue())
+        if (params.HasValuePairingInstr())
         {
-            sprintf(txtPairingInstr, "PI=%s", params.GetPairingInstr().Value());
+            sprintf(txtPairingInstr, "PI=%s", params.GetPairingInstr());
```

Comment:
Please use snprintf here and above in every instance since buffer size is known and can be safely capped

Reply:
Appears resolved